RHDHBUGS-3223: Orchestrator docs Remove insecure config examples and improve inherit explanation placement (#2286)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/proc-enable-the-orchestrator-plugins-using-the-operator.adoc

.Procedure

. To enable the Orchestrator plugins with default settings, set `disabled: false` for the corresponding packages.
+

> The `{{inherit}}` attribute in the package field automatically resolves to the plugin version and default plugin configuration for the version of {product-very-short} that you are using.

Reviewer: ```suggestion
The `{{inherit}}` attribute in the package field automatically resolves to the default plugin version and configuration for your version of {product-very-short}.
```

+
[NOTE]
====

> . The `{{inherit}}` attribute in the package versions automatically resolves to the {product-version} version and default plugin configuration provided by the Operator. This simplifies plugin management by eliminating the need to manually update versions when upgrading {product-very-short}.

Reviewer 2: Same point as in my previous comment.

+
[NOTE]
====

> If you need a specific plugin version, replace `{{inherit}}` with the version tag, for example: `1.3.0`

Reviewer: ```suggestion
If you need a specific plugin version, replace `{{inherit}}` with the version tag, such as `1.3.0`
```

> The `ref: sonataflow` field installs the OpenShift Serverless and OpenShift Serverless Logic resources when using the Operator.

Reviewer: ```suggestion
When you use the Operator, the `ref: sonataflow` field installs the OpenShift Serverless and OpenShift Serverless Logic resources.
```

====
+

> The following example shows a complete configuration of the Orchestrator plugin `dynamic-plugins` ConfigMap:

Reviewer: ```suggestion
The following example shows a complete configuration of the Orchestrator plugin `dynamic-plugins` config map:
```

          disabled: false
        - package: "oci://registry.access.redhat.com/rhdh/red-hat-developer-hub-backstage-plugin-orchestrator-form-widgets:{{inherit}}"
          disabled: false
----

> . Configure backend authentication by creating a secret containing a securely generated `BACKEND_SECRET` value:

Reviewer 3: We need to also address the _why_ here. My understanding is that the _why_ is to allow the Orchestrator components to communicate with RHDH's API, but I don't think it's strictly required unless an orchestrator workflow uses it? CC @pkliczewski

Reviewer 4: we need it to call notifications or scaffolder plugin. If they are not used in a workflow the secret will not be used.

          disabled: false
        - package: "oci://registry.access.redhat.com/rhdh/red-hat-developer-hub-backstage-plugin-orchestrator-form-widgets:{{inherit}}"
          disabled: false
----

. (Optional) Configure backend authentication to allow Orchestrator components to communicate with the {product-very-short} API.
+

> This step is only required when your workflow definitions include API calls to {product-very-short} plugins, such as the notifications or scaffolding plugins.

Reviewer: ```suggestion
This step is required only when your workflow definitions include API calls to {product-very-short} plugins, such as notifications or scaffolding.
```

>     dynamicPluginsConfigMapName: orchestrator-plugin

Reviewer 2: I think these lines with the Backstage CR can be configured from this example, as there is a step below showing how to configure the Backstage CR.

<img width="850" height="491" alt="Image" src="https://github.com/user-attachments/assets/7f95cae9-4bbe-4e51-941e-06e43dbb1996" />

As this full example is missing the `app-config-rhdh` ConfigMap, which is actually depicted in step 2 just below. And it looks like this complete example is meant to be an example of the dynamic-plugins ConfigMap only, IMO.

metadata:
  name: backend-auth-secret
stringData:

>   # Generate a secure random value using:

Reviewer: ```suggestion
  # Run the following command to generate a secure random value:
```

  # node -p 'require("crypto").randomBytes(24).toString("base64")'
  BACKEND_SECRET: "<GENERATED_VALUE>"
----
+
[IMPORTANT]
====

> Replace `<GENERATED_VALUE>` with a securely generated random token. Never use example or placeholder values in production environments.

Reviewer: ```suggestion
Replace `<GENERATED_VALUE>` with a securely generated random token. Do not use example or placeholder values in production environments.
```

====

> . If you configured backend authentication in the previous step, update your {product-custom-resource-type} CR to reference the secret:

Reviewer: ```suggestion
. If you have configured backend authentication, update your {product-custom-resource-type} CR to reference the secret:
```
